internal/snykclient/snykclient.go

```diff
@@ -112,15 +116,39 @@ func (t *SBOMTest) GetStatus(ctx context.Context, errFactory *errors.ErrorFactor
 		return SBOMTestStatusFinished, nil
 	}
 
+	body, err := io.ReadAll(resp.Body)
```

Comment:
I would consider creating a json.Decoder, and making us of that on line 128.

Reply:
that was the initial implementation, but when you get down to L148 (fallback) we unmarshal once more. at this point the `resp.body` stream would have been consumed, so we need to assign it to this var instead and use `json.Unmarshal`. could've been my ignorance of some golang features, but this was the best i could come up with.

Reply:
Ah then I know, that's a good reason for it.

I think that's the intention of GetBody (even if it's a bit hidden in the ToC of the docs)

```
// GetBody defines an optional func to return a new copy of
	// Body. It is used for client requests when a redirect requires
	// reading the body more than once. Use of GetBody still
	// requires setting Body.
	//
	// For server requests, it is unused.
	GetBody func() ([io](https://pkg.go.dev/io).[ReadCloser](https://pkg.go.dev/io#ReadCloser), [error](https://pkg.go.dev/builtin#error))
```

Reply:
Never mind, reading it again I'm probably wrong. It's for the client not the server.

Reply:
another alternative might be `io.TeeReader(` but your implementation works, so run with that one :)